### PR TITLE
Add eslint-plugin-ramda

### LINKED ISSUE #306: Add eslint-plugin-ramda
See: https://github.com/ramda/eslint-plugin-ramda

Adds some nice rules, though possibly a few we will want to disable.

### PR DESCRIPTION
**Pull request by Undistraction**
Code Climate have now added support for eslint-plugin-ramda.

See https://github.com/codeclimate/codeclimate-eslint/issues/369#issuecomment-362718696

Closes #306

### CODE REVIEW

**Comment by Undistraction** [comment]
Still not working. I've flagged it on the linked issue.

**Comment by Undistraction** [comment]
@charon All passing now

**Comment by Undistraction** [comment]
@char0n  All passing now. Should be good to merge.

**Comment by char0n** [comment]
Cool

File: src/internal/polyfills/Number.MAX_SAFE_INTEGER.js [hunk 1]
```diff
@@ -1,6 +1,3 @@
-import { or } from 'ramda';
-
-
-const MAX_SAFE_INTEGER = or(Number.MAX_SAFE_INTEGER, (2 ** 53) - 1);
+const MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || (2 ** 53) - 1;
```
Comment (char0n) [inline src/internal/polyfills/Number.MAX_SAFE_INTEGER.js]: Would use parenthesis to explicitly state the associations of operands
```js
const MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || ((2 ** 53) - 1)

File: src/internal/polyfills/Number.MIN_SAFE_INTEGER.js [hunk 2]
```diff
@@ -1,6 +1,3 @@
-import { or } from 'ramda';
-
-
-const MIN_SAFE_INTEGER = or(Number.MIN_SAFE_INTEGER, -(2 ** 53) - 1);
+const MIN_SAFE_INTEGER = Number.MIN_SAFE_INTEGER || -(2 ** 53) - 1;
```
Comment (char0n) [inline src/internal/polyfills/Number.MIN_SAFE_INTEGER.js]: Would use parenthesis to explicitly state the associations of operands
```js
const MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || (-(2 ** 53) - 1)

Comment (Undistraction) [reply src/internal/polyfills/Number.MIN_SAFE_INTEGER.js]: Tbh I'm so used to prettier purging my files of extra parens that I it doesn't bother me. Sooner or later we'll start using it for this project and they'll be gone anyway. 

Not saying I disagree with you (I'v actually been complaining about its handling of parens for ages - [see](https://github.com/prettier/prettier/issues/187#issuecomment-282741690) my comment towards the end), but using prettier is such a huge win in terms of time that it's worth the small bumps.

So doesn't bother me but happy to change if you would like.

**Comment by char0n** [comment]
I've create https://github.com/char0n/ramda-adjunct/issues/336 for prettier integration. Wanted to try him for quite some time, this is good oportunity. Don't worry about the parents, I'll add them.

**Comment by Undistraction** [comment]
@char0n WIth prettier you have to just accept that there are very few config options. That is actually the best thing about it, but if you are used to finessing how you like things to look it will be a hard at first. I honestly think it is the single biggest time-saver you can use in writing JS. So much less distraction from silly little decisions about formatting. It's very fast too.

**Pull request merged**